corehq/form_processor/parsers/form.py

```diff
@@ -185,6 +186,13 @@ def _handle_duplicate(new_doc):
 
     new_md5 = new_doc.xml_md5()
 
+    is_icds = settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS
+    if is_icds and new_doc.metadata.deviceID == existing_doc.metadata.deviceID:
```

Comment:
Is there a reason this check is done after `existing_md5` and `new_md5` are calculated? It does not appear to use them, and those are relatively expensive calculations.

Reply:
85dffb2f437